libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSDKWKProcessPoolFactory.m

```diff
@@ -0,0 +1,46 @@
+/*
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
+
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#import <WebKit/WebKit.h>
+#import "SFSDKWKProcessPoolFactory.h"
+
+@implementation SFSDKWKProcessPoolFactory
+
+static WKProcessPool *_processPool = nil;
+
++ (void)initialize {
+    if (self == [SFSDKWKProcessPoolFactory class]) {
+        self.sharedProcessPool = [[WKProcessPool alloc] init];
+    }
+}
+
++ (WKProcessPool *)sharedProcessPool {
+    return _processPool;
+}
+
++ (void)setSharedProcessPool:(WKProcessPool *)sharedProcessPool {
+    if (_processPool != sharedProcessPool) {
+        _processPool = sharedProcessPool ?: [[WKProcessPool alloc] init];
```

Comment:
We had some internal discussion about this from before, and I'd like to suggest we implement a slightly different pattern.  Namely, this is a pattern that would more conventionally be flipped to the getter, where, if `_processPool` is `nil`, it would return a new `WKProcessPool` instance.  That would also alleviate the need for the `initialize` method.

It just cleans up a lot of code, and follows a more conventional pattern.